File: PracticeRecyclerView/app/src/main/java/com/example/practicerecyclerview/MyAdapter.java
package com.example.practicerecyclerview;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.w3c.dom.Text;

import java.util.List;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    List<MyListData> myListData;

    public MyAdapter(List<MyListData> myListData) {
        this.myListData = myListData;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_layout, viewGroup, false);

        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, int position) {
        MyListData listData = myListData.get(position);
        myViewHolder.myListDataTextView.setText(listData.getMyListData());
    }

    @Override
    public int getItemCount() {
        return myListData.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        TextView myListDataTextView;
        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            myListDataTextView = itemView.findViewById(R.id.myListDataTextView);

        }
    }
}

File: KotlinRetrofitRxJavaRecyclerView/app/src/main/java/com/calvert/kotlinretrofitrxjavarecyclerview/adapter/MyAdapter.kt
package com.calvert.kotlinretrofitrxjavarecyclerview.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.calvert.kotlinretrofitrxjavarecyclerview.R
import com.calvert.kotlinretrofitrxjavarecyclerview.model.Post
import kotlinx.android.synthetic.main.list_post.view.*

class MyAdapter(val context: Context, val postList: List<Post>) : RecyclerView.Adapter<MyAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.list_post, parent, false)
        return MyViewHolder(view)
    }

    override fun getItemCount(): Int {
        return postList.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.bindValues(postList.get(position))
    }


    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bindValues(post: Post){
            itemView.user_id_tv.text = post.userId.toString()
            itemView.id_tv.text = post.id.toString()
            itemView.title_tv.text = post.title
            itemView.body_tv.text = post.body
        }
    }

}